lib/vector-core/src/event/mod.rs

```diff
@@ -39,49 +39,50 @@ pub enum Event {
 }
 
 impl Event {
+    #[must_use]
     pub fn new_empty_log() -> Self {
         Event::Log(LogEvent::default())
     }
 
     pub fn as_log(&self) -> &LogEvent {
         match self {
             Event::Log(log) => log,
-            _ => panic!("Failed type coercion, {:?} is not a log event", self),
+            Event::Metric(_) => panic!("Failed type coercion, {:?} is not a log event", self),
```

Comment:
What is the motivation here? We will eventually add a trace variant to `enum Event` and then we will have to undo all of these.

Reply:
Yeah, I agree with this. I think `if let Some()` would be clearer and also satisfy clippy.

Reply:
Reversed in d949fa1